projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -1811,9 +1814,9 @@ function __construct( $attributes, $content = null ) {
 	function parse_content( $content ) {
 		if ( is_null( $content ) ) {
 			$this->body = null;
+		} else {
+			$this->body = do_shortcode( $content );
 		}
-
-		$this->body = do_shortcode( $content );
```

Comment:
I'm a bit confused by this change. Was this causing any kind of issue before?

Reply:
The top of `do_shortcode()` looks like this:
```php
 * @param string $content     Content to search for shortcodes.
 * @param bool   $ignore_html When true, shortcodes inside HTML elements will be skipped.
 *                            Default false.
 * @return string Content with shortcodes filtered out.
 */
function do_shortcode( $content, $ignore_html = false ) {
    global $shortcode_tags;

    if ( false === strpos( $content, '[' ) ) {
        return $content;
    }
```
If we pass null, that `strpos()` call raises a deprecation warning in PHP 8.1. Since `do_shortcode()` is declared as taking a string, not "string|null" or the like, it seems best to just avoid calling it with null in the first place.

A string cast on `$content` here would be a behavior change, as previously `do_shortcode()` would just return the null `$content` while casting `$content` to string would result in it returning that empty string instead.